### PR TITLE
IPv6 to Ipv4 Fallback support in parodus

### PR DESCRIPTION
**Pull request by selvamKrish**

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -280,7 +290,13 @@ int createNopollConnection(noPollCtx *ctx)
 					}
 				}			
 			}
+
 			initial_retry = true;
+			// If IPv6 conn failed to connect then fallback to IPv4 conn or vice-versa
+			if(FLAGS_IPV6_ONLY == (FLAGS_IPV6_IPV4 & fallback))
```
Comment (shilpa24balaji) [inline src/connection.c]: Make this if else block a function toggleIPFlag(int *ptr)

File: src/connection.c [hunk 2]
```diff
@@ -353,14 +369,17 @@ static noPollConn * nopoll_tls_common_conn (noPollCtx  * ctx,char * serverAddr,c
...
-                ParodusInfo("Ipv6 connection failed. Try connecting with Ipv4 mode \n");
-                opts = createConnOpts(extra_headers, true);
-                connection = nopoll_conn_tls_new (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
-            }
+			if(FLAGS_IPV6_ONLY == (FLAGS_IPV6_IPV4 & fallback))
+			{
+				ParodusInfo("Try connecting with Ipv6 mode\n");
+				connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
+			}
+			if(!nopoll_conn_is_ok (connection) || FLAGS_IPV4_ONLY == (FLAGS_IPV6_IPV4 & fallback))
```
Comment (shilpa24balaji) [inline src/connection.c]: I guess we should toggle fallback if this condition becomes true 
!nopoll_conn_is_ok (connection)

File: src/connection.c [hunk 3]
```diff
@@ -353,14 +369,17 @@ static noPollConn * nopoll_tls_common_conn (noPollCtx  * ctx,char * serverAddr,c
...
         } else {
-            ParodusInfo("Try connecting with Ipv6 mode\n");
-            connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
-            if(connection == NULL)
-            {
-                ParodusInfo("Ipv6 connection failed. Try connecting with Ipv4 mode \n");
-                opts = createConnOpts(extra_headers, true);
-                connection = nopoll_conn_tls_new (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
-            }
+			if(FLAGS_IPV6_ONLY == (FLAGS_IPV6_IPV4 & fallback))
```
Comment (shilpa24balaji) [inline src/connection.c]: Update get_parodus_cfg()->flags to FLAGS_IPv6_IPv4

File: src/connection.c [hunk 4]
```diff
@@ -353,14 +369,17 @@ static noPollConn * nopoll_tls_common_conn (noPollCtx  * ctx,char * serverAddr,c
             ParodusInfo("Connecting in Ipv6 mode\n");
             connection = nopoll_conn_tls_new6 (ctx, opts,serverAddr,serverPort,NULL,get_parodus_cfg()->webpa_path_url,NULL,NULL);
         } else {
```
Comment (shilpa24balaji) [inline src/connection.c]: Update get_parodus_cfg()->flags to FLAGS_IPv6_IPv4

Comment (selvamKrish) [reply src/connection.c]: To update get_parodus_cfg()->flags to FLAGS_IPv6_IPv4 , we need ParodusCfg struct in nopoll_tls_common_conn. is that really required to update get_parodus_cfg()->flags with FLAGS_IPv6_IPv4 ? if yes then we need to change the function prototype declaration for createNopollConnection and nopoll_tls_common_conn to pass ParodusCfg as new args.

Comment (shilpa24balaji) [reply src/connection.c]: Not required

File: src/connection.c (hunk 1 above)
Comment (shilpa24balaji) [reply src/connection.c]: Why is this if else not replaced by toggleIPFlag?

**Comment by selvamKrish** [comment]
Given PR https://github.com/Comcast/parodus/pull/173

**Pull request closed without merging**